test/libyul/functionSideEffects/structures.yul

```diff
@@ -31,10 +31,10 @@
     }
 }
 // ----
-// : invalidatesStorage, invalidatesMemory
-// f: sideEffectFreeIfNoMSize
-// g: invalidatesStorage, invalidatesMemory
-// h: invalidatesStorage, invalidatesMemory
-// i: invalidatesStorage
-// r: movable, sideEffectFree, sideEffectFreeIfNoMSize
-// t: invalidatesMemory
+// : writes storage, writes memory
+// f: movable apart from effects, can be removed if no msize, reads memory
+// g: writes storage, writes memory
+// h: writes storage, writes memory
+// i: writes storage
+// r: movable, movable apart from effects, can be removed, can be removed if no msize
```

Comment:
Which effects?

Reply:
This means the flag `movableApartFromEffect` is true. So it can be "moved relative to" under certain combinations of `SideEffects::storage, SideEffects::memory` and `SideEffects::otherState`. The code for that is in the LoopInvariantCodeMotionPR https://github.com/ethereum/solidity/blob/d902ac90cd61951e77435f0b9066d8bc4d86c01a/libyul/optimiser/Semantics.h#L68